ecosystem/ffmpeg_plugin/kahawai_enc.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * Kahawai raw video muxer
+ * Copyright (c) 2023 Intel
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <mtl/st_convert_api.h>
+#include <mtl/st_pipeline_api.h>
+
+#include "libavformat/avformat.h"
+#include "libavformat/internal.h"
+#include "libavutil/imgutils.h"
+#include "libavutil/opt.h"
+#include "libavutil/parseutils.h"
+#include "libavutil/pixdesc.h"
+#include <unistd.h>
+#include "kahawai_common.h"
+
+typedef struct KahawaiMuxerContext {
+  const AVClass *class; /**< Class for private options. */
+
+  char *port;
+  char *local_addr;
+  char *dst_addr;
+  int udp_port;
+  int fb_cnt;
+  int session_cnt;
+
+  int width;
+  int height;
+  enum AVPixelFormat pixel_format;
+  AVRational framerate;
+  mtl_handle dev_handle;
+  st20p_tx_handle tx_handle;
+
+  pthread_cond_t get_frame_cond;
+  pthread_mutex_t get_frame_mutex;
+
+  int64_t frame_counter;
+  struct st_frame *frame;
+  size_t output_frame_size;
+} KahawaiMuxerContext;
+
+static mtl_handle shared_st_handle = NULL;
+static unsigned int active_session_cnt = 0;
+static int64_t frame_tx_completed = 0;
+static bool tx_completed = false;
+
+extern mtl_handle shared_st_handle;
+extern unsigned int active_session_cnt;
```

Comment:
Confused that shared_st_handle is declared before already. Same for active_session_cnt.
```
static mtl_handle shared_st_handle = NULL;
static unsigned int active_session_cnt = 0;
```